src/snowflake/cli/_plugins/nativeapp/entities/application_package.py

```diff
@@ -439,12 +442,24 @@ def deploy(
         if validate:
             cls.validate_setup_script(
                 console=console,
+                project_root=project_root,
+                deploy_root=deploy_root,
+                bundle_root=bundle_root,
+                generated_root=generated_root,
+                artifacts=artifacts,
                 package_name=package_name,
                 package_role=package_role,
+                package_distribution=package_distribution,
+                prune=prune,
+                recursive=recursive,
+                paths=paths,
                 stage_fqn=stage_fqn,
+                package_warehouse=package_warehouse,
+                post_deploy_hooks=post_deploy_hooks,
+                package_scripts=package_scripts,
+                policy=policy,
                 use_scratch_stage=False,
                 scratch_stage_fqn="",
-                deploy_to_scratch_stage_fn=lambda *args: None,
```

Comment:
how is this translated? Do we skip because scratch_stage_fqn is empty? if so, should we use None instead to be more explicit?

Reply:
`use_scratch_stage=False`, so this doesn't run: https://github.com/snowflakedb/snowflake-cli/pull/1621/files/647e730c5df8cff0bd20b7ee849e045513041a18#diff-65efa774f47f1a181d9d65e29c111f229d7e6bd7eef48e70ec5fc81042bf3beeR1241-R1243, meaning that we run validation on the main app stage that was already deployed. I can change this to `None`, but I'm not a fan of `Optional[str]` when an empty string means the same thing (since in this case an empty stage name isn't valid)